main.py

```diff
@@ -26,9 +26,9 @@ def load_data(chart_id):
                                                 "message": "No such chart"}}
         return response, 404
 
-@app.route("/clustering/<algorithm>/<similarity>/<encoding>/<outlier>/<chart_id>")
-@app.route("/clustering/<algorithm>/<similarity>/<encoding>/<outlier>/<chart_id>/<key>")
-def cluster(algorithm, similarity, encoding, outlier, chart_id, key=None):
+@app.route("/clustering/<algorithm>/<similarity>/<encoding>/<outlier>/<rep>/<chart_id>")
+@app.route("/clustering/<algorithm>/<similarity>/<encoding>/<outlier>/<rep>/<chart_id>/<key>")
```

Comment:
Why are there two calls to @app.route here?

Reply:
Having two routes allows both of them to map to this function. Since key is optional, I want the route with it and the route without out it to map to this.